File: 0509-fibonacci-number/0509-fibonacci-number.js
/**
 * @param {number} n
 * @return {number}
 */
var fib = function (n) {
    if (n == 0) {
        return 0;
    } else if (n == 1) {
        return 1;
    }

    let firstTerm = 0;
    let secondTerm = 1;

    for (let i = 1; i < n; i++) {
        let thirdTerm = firstTerm + secondTerm;
        firstTerm = secondTerm;
        secondTerm = thirdTerm;
    }

    return secondTerm;
};

File: 0509-fibonacci-number/0509-fibonacci-number.c


int fib(int n) {
    if (n == 0) {
        return 0;
    } else if (n == 1) {
        return 1;
    }

    int firstTerm = 0;
    int secondTerm = 1;

    for (int i = 0; i < n - 1; i++) {
        int temp = secondTerm;
        secondTerm = secondTerm + firstTerm;
        firstTerm = temp;
    }

    return secondTerm;
}